Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/SharedLogic/ScheduledNotifications.hs

```diff
@@ -0,0 +1,50 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module SharedLogic.ScheduledNotifications where
+
+import qualified Domain.Types.Booking as DB
+import qualified Domain.Types.Person as DP
+import qualified Domain.Types.RideRelatedNotificationConfig as DRN
+import Kernel.External.Types (SchedulerFlow)
+import Kernel.Prelude
+import Kernel.Types.Id
+import Kernel.Utils.Common
+import Lib.Scheduler.JobStorageType.SchedulerType (createJobIn)
+import SharedLogic.Allocator
+
+pushReminderUpdatesInScheduler :: (MonadFlow m, SchedulerFlow r, EsqDBFlow m r, CacheFlow m r) => DB.Booking -> Id DP.Person -> DRN.RideRelatedNotificationConfig -> m ()
+pushReminderUpdatesInScheduler booking driverId DRN.RideRelatedNotificationConfig {..} = do
+  now <- getCurrentTime
+  let eventTime = case timeDiffEvent of
+        DRN.RIDE_ASSIGNED -> booking.updatedAt -- gives the time of TRIP_ASSIGNED
+        DRN.PICKUP_TIME -> booking.startTime -- scheduled time
+        _ -> now -- this will not get scheduled
+  let currentTimeDiffFromScheduledTime = diffUTCTime eventTime now
```

Comment:
Did you test it? It should be `diffUTCTime now eventTime`. But this logic is incorrect.